src/api/factories/EscrowFactory.ts

```diff
@@ -0,0 +1,102 @@
+import * as _ from 'lodash';
+import { inject, named } from 'inversify';
+import { Logger as LoggerType } from '../../core/Logger';
+import { Types, Core, Targets } from '../../constants';
+import { Escrow } from '../models/Escrow';
+import { EscrowLockMessage } from '../messages/escrow/EscrowLockMessage';
+import { EscrowRefundMessage } from '../messages/escrow/EscrowRefundMessage';
+import { EscrowReleaseMessage } from '../messages/escrow/EscrowReleaseMessage';
+
+export class EscrowFactory {
+
+    public log: LoggerType;
+
+    constructor(
+        @inject(Types.Core) @named(Core.Logger) public Logger: typeof LoggerType
+    ) {
+        this.log = new Logger(__filename);
+    }
+
+
+    /**
+     * data:
+     * escrow: object
+     * address: object
+     * listing: string
+     * nonce: string
+     * memo: string
+     *
+     * @param data
+     * @returns {Promise<Escrow>}
+     */
+    public getLockMessage(data: EscrowLockMessage): Promise<Escrow> {
+        return {
```

Comment:
input EscrowLockMessage? output Promise<Escrow>?
use one get method, input should be XRequest or multiple params, output should maybe be something like EscrowMessageInterface, being able to return the correct type of EscrowMessage.